Sources/Switchcraft/Switchcraft.swift

```diff
@@ -52,21 +52,59 @@ public class Switchcraft {
      */
     public init(config: Config) {
         guard !config.endpoints.isEmpty else {
-            fatalError("Switchcraft.setup called with no endpoints set. You probably didn't mean to do that.")
+            fatalError("Switchcraft.init called with no endpoints set. You probably didn't mean to do that.")
+        }
+        
+        guard config.endpoints.indices.contains(config.defaultEndpointIndex) else {
+            fatalError("Switchcraft.init called with a default endpoint index that's out of bounds. You probably didn't mean to do that.")
         }
 
         self.config = config
-
-        let endpoints = config.endpoints
-        if !endpoints.indices.contains(config.defaultEndpointIndex) {
-            debugPrint("`defaultEndpointIndex` was invalid, reverting to the first endpoint. ")
-            self.config.defaultEndpointIndex = 0
+        let newEndpoints = config.endpoints
+        
+        /*
+         * Set current endpoint if it's not found (could be the first time, could be a different key).
+         * If it's a different key, which is considered as a deprecation of the current UserDefaults,
+         * then it's a full reset to the config's default endpoint,
+         * regardless of whether the list of endpoints remain the same or not.
+         */
+        guard let currentEndpoint = endpoint else {
+            selectAndCache(endpoint: newEndpoints[config.defaultEndpointIndex])
+            return
         }
-
-        // If there's no endpoint stored, store the default one
-        if endpoint == nil {
-            selected(endpoint: endpoints[config.defaultEndpointIndex])
+        
+        /*
+         * If current endpoint exists, we also need to evaluate whether the config has updated just the url or just the name for this endpoint.
+         * If only the url or the name has changed, it is essentially the same record but updated. Therefore, we don't actually want
+         * to reset the current selection back to the default endpoint, but rather just update the current endpoint in UserDefaults,
+         * so that in case the default endpoint is different, the user does not need to repick the endpoint just because the name or the url got updated.
+         */
+        if let newEndpointWithSameName = config.endpoints.first(where: { $0.name == currentEndpoint.name }) {
+            if currentEndpoint.url != newEndpointWithSameName.url {
```

Comment:
Do you think it makes sense to leave these two if statements nested, or maybe it's easier to parse if they're combined?

Reply:
@brendanlensink good point and I thought about that too! Except even on a big screen, the line is long enough that it "appears nested" too after wrapping 😂 And then it's wrapping vs 1 indentation – no clear winner imo 😮‍💨

But if you prefer 1 line, let me know and I can make the change, since I am indifferent in this particular case and the scale can be tipped with just 1 preference!

Reply:
I'm partial to combining the if statements, even if you have to split it over two lines, but really it's not a big deal either way.

```
       if let newEndpointWithSameName = config.endpoints.first(where: { $0.name == currentEndpoint.name }),
                currentEndpoint.url != newEndpointWithSameName.url {
```